### DISCUSSION TITLE
Can we send current data to inverter?

### DISCUSSION THREAD

**Discussion by Blackra1n5** [wlcrs/huawei_solar#397] [General]
We kinda have a weird setup where 2 buildings (on the same meter) are each supplied directly from the pole. 

I am wonder if I can have an ESP32 which is hooked up to a CT on the other building aggregate the power of both buildings and send them to the Huawei inverter (similar to the "Huawei Smart Meters")?

Thank you

**Comment by g3r0nim00** [comment]
That would be a question for Huawei/..., it's not related to this add-on which is only communicating with the inverter (and attached devices).

**Reply by Blackra1n5** [reply]
Is the register which data is being sent from the smart meter to the inverter accessible via this API?

**Reply by gaborvar** [reply]
Whether the Huawei inverter can accept meter data through an API other than the Huawei smart meter Modbus/RS484 interface would be a significant factor to several optimization scenarios. Several intelligent devices (e.g. EV chargers, etc.) spring up and each require a smart meter at the grid connection point. I hear that several buildings duplicate and triplicate quite similar smart meters at the same location just because vendors do not talk to each other and accept each other's readings.  Support considerations aside, an API to push meter data to the Huawei system would reduce cost and complexity.  DSMR P1 data is also a good candidate as source, although P1 implementations deviate and may prove to be inadequate.  

Release notes https://support.huawei.com/enterprise/en/fusionsolar/sdongle-pid-23826585 of the Huawei dongle firmware may include useful background but I did not find a direct answer:

**Comment by apos-t** [comment]
Any Updates? I've a shelly Pro 3EM and would like to use it instead of the dts666-H smartmeter. Found this modbus documentation https://www.photovoltaikforum.com/core/attachment/251184-solar-inverter-modbus-interface-definitions-pdf/

or is there an other way to control the charge / discharge of the LUNA in Home Assistant.
